runtests.py

```diff
@@ -183,7 +183,7 @@ def main(argv):
     if args.shell:
         shell = os.environ.get('SHELL', 'sh')
         print("Spawning a Unix shell...")
-        os.execv(shell, [shell] + extra_argv)
+        subprocess.check_call([shell] + extra_argv)
         sys.exit(1)
```

Comment:
Fixed. Also added another commit to make it launch a more appropriate shell on windows